pkg/aws/policies.go

```diff
@@ -743,16 +752,34 @@ func (c *awsClient) listPolicies(role *iam.Role) ([]Policy, error) {
 	return policies, nil
 }
 
-func (c *awsClient) ListAccountRoles(version string) ([]Role, error) {
-	accountRoles := []Role{}
-	roles, err := c.ListRoles()
+func (c *awsClient) GetAccountRoleByArn(arn string) (*Role, error) {
+	role, err := c.GetRoleByARN(arn)
 	if err != nil {
-		return accountRoles, err
+		return nil, err
 	}
+
+	roles := []*iam.Role{role}
+	accountRoles, err := c.mapToAccountRoles("", roles)
```

Comment:
Should we refactor only the part that handles a single role to `mapToAccountRole`? That way we don't need `roles := []*iam.Role{role}`, and can reuse the function inside a loop in `ListAccountRoles`. WDYT?

Reply:
Yep, good call. I'll do that 👍